compiler/stable_mir/src/fold.rs

```diff
@@ -15,6 +18,9 @@ pub trait Folder: Sized {
     fn fold_const(&mut self, c: &Const) -> ControlFlow<Self::Break, Const> {
         c.super_fold(self)
     }
+    fn visit_reg(&mut self, reg: &Region) -> ControlFlow<Self::Break, Region> {
```

Comment:
lol uhm... I messed up here. Please use `fold_reg` and change `visit_ty` to `fold_ty` while you're at it.